src/internal/github/client.go

```diff
@@ -2,17 +2,77 @@ package github
 
 import (
 	"context"
-	"github.com/shurcooL/githubv4"
-	"golang.org/x/oauth2"
+	"fmt"
+	"log/slog"
 	"net/http"
+	"os"
+	"time"
+
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/time/rate"
 )
 
-func NewGitHubClient(ctx context.Context, token string) *githubv4.Client {
-	return githubv4.NewClient(getGithubOauth2Client(ctx, token))
+func EnvAuthToken() (string, error) {
+	token := os.Getenv("GH_TOKEN")
+	if token == "" {
+		return "", fmt.Errorf("Expected $GH_TOKEN to be set, unable to authenticate with GitHub")
+	}
+	return token, nil
+
+}
+
+type Client struct {
+	ctx        context.Context
+	log        *slog.Logger
+	httpClient *http.Client
+	ghClient   *githubv4.Client
+}
+
+func NewClient(ctx context.Context, log *slog.Logger, token string) Client {
```

Comment:
I'd rather move github.Client to an interface when we get to that level of testing.  That would allow us to easily create a mock that would implement the github.Client interface without hacking an existing struct.